jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,444 @@
+
+  if (size == 0)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_string_optional (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (jerry_value_is_undefined (js_arg))
+  {
+    return jerry_create_undefined ();
+  }
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (to_string,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  jerry_release_value (to_string);
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_string (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (to_string,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  jerry_release_value (to_string);
+
+  return jerry_create_undefined ();
+}
+
+/* function transform functions. */
+
+jerry_value_t
+jerryx_arg_transform_function (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_function (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a function.");
+  }
+
+  jerry_value_t *func_p = jerryx_arg_iterator_get_dest (iterator_p);
+  *func_p = jerry_acquire_value (js_arg);
```

Comment:
`jerry_release_value(*func_p)` is a dangerous operation.

if users didn't assign the initial value of the jerry_value_t, e.g.
```
jerry_value_t my_func;
//... pass the &my_func to jerryx_arg_t, then do the validation&assign
```
the initial value of my_func could be any value, which contain the address info and jerry_release_value(*func_p) may change value of  that address.